pkg/client/testclient/fake.go

```diff
@@ -38,6 +41,9 @@ func NewSimpleFake(objects ...runtime.Object) *Fake {
 // Invokes registers the passed fake action and reacts on it if a ReactFn
 // has been defined
 func (c *Fake) Invokes(action FakeAction, obj runtime.Object) (runtime.Object, error) {
+	c.Lock()
+	defer c.Unlock()
+
 	c.Actions = append(c.Actions, action)
 	if c.ReactFn != nil {
 		return c.ReactFn(ktestclient.FakeAction(action))
```

Comment:
> Actually calling Invokes inside Invokes wouldn't be smart

Maybe not, but the test client shouldn't deadlock if you do that in a custom ReactFn